halo2-base/src/gates/tests/range_gate_test.rs

```diff
@@ -0,0 +1,114 @@
+}
+
+#[test_case((Fr::zero(), 1); "check_less_than_safe() pos")]
+fn test_check_less_than_safe<F: ScalarField>(inputs: (F, u64)) {
+    let mut builder = GateThreadBuilder::mock();
+    let ctx = builder.main(0);
+    let chip = RangeChip::default(3);
+    let a = ctx.assign_witnesses([inputs.0])[0]; 
+    chip.check_less_than_safe(ctx, a, inputs.1);
+    // auto-tune circuit
+    builder.config(11, Some(9));
+    // create circuit
+    let circuit = RangeCircuitBuilder::mock(builder);
+    MockProver::run(11 as u32, &circuit, vec![]).unwrap().assert_satisfied()
+}
+
+#[test_case((Fr::zero(), 1); "check_big_less_than_safe() pos")]
+fn test_check_big_less_than_safe<F: ScalarField + BigPrimeField>(inputs: (F, u64)) {
+    let mut builder = GateThreadBuilder::mock();
+    let ctx = builder.main(0);
+    let chip = RangeChip::default(3);
+    let a = ctx.assign_witnesses([inputs.0])[0]; 
+    chip.check_big_less_than_safe(ctx, a, BigUint::from(inputs.1));
+    // auto-tune circuit
+    builder.config(11, Some(9));
+    // create circuit
+    let circuit = RangeCircuitBuilder::mock(builder);
+    MockProver::run(11 as u32, &circuit, vec![]).unwrap().assert_satisfied()
+}
+
+#[test_case(([0, 1].map(Fr::from).map(Witness), 3) => Fr::from(1) ; "is_less_than() pos")]
+fn test_is_less_than<F: ScalarField>(inputs: ([QuantumCell<F>; 2], usize)) -> F {
+    let mut builder = GateThreadBuilder::mock();
+    let ctx = builder.main(0);
+    let chip = RangeChip::default(3);
+    let a = chip.is_less_than(ctx, inputs.0[0], inputs.0[1], inputs.1);
+    *a.value()
+}
+
+#[test_case((Fr::zero(), 3) => Fr::from(1) ; "is_less_than_safe() pos")]
+fn test_is_less_than_safe<F: ScalarField>(inputs: (F, u64)) -> F {
+    let mut builder = GateThreadBuilder::mock();
+    let ctx = builder.main(0);
+    let chip = RangeChip::default(3);
+    let a = ctx.assign_witnesses([inputs.0])[0]; 
+    let b = chip.is_less_than_safe(ctx, a, inputs.1);
+    *b.value()
+}
+
+#[test_case((Fr::zero(), 3) => Fr::from(1) ; "is_big_less_than_safe() pos")]
+fn test_is_big_less_than_safe<F: ScalarField + BigPrimeField>(inputs: (F, u64)) -> F {
+    let mut builder = GateThreadBuilder::mock();
+    let ctx = builder.main(0);
+    let chip= RangeChip::default(3);
+    let a = ctx.assign_witnesses([inputs.0])[0]; 
+    let b = chip.is_big_less_than_safe(ctx, a, BigUint::from(inputs.1));
+    *b.value()
+}
+
+#[test_case((Witness(Fr::one()), 1, 2) => (Fr::one(), Fr::zero()) ; "div_mod() pos")]
+fn test_div_mod<F: ScalarField + BigPrimeField>(inputs: (QuantumCell<F>, u64, usize)) -> (F, F) {
+    let mut builder = GateThreadBuilder::mock();
+    let ctx = builder.main(0);
+    let chip = RangeChip::default(3);
+    let a = chip.div_mod(ctx, inputs.0, BigUint::from(inputs.1), inputs.2);
+    (*a.0.value(), *a.1.value())
+}
+
+#[test_case((Fr::one(), 2) => Fr::from(0) ; "get_last_bit() pos")]
+fn test_get_last_bit<F: ScalarField>(inputs: (F, usize)) -> F {
```

Comment:
A plug for more randomized testing - `get_last_bit` is currently broken, so tests should *not* pass :)